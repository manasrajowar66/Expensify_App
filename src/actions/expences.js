import {v4 as uuid} from 'uuid';

export const addExpence = ({describtion = '', note = '', amount = 0 ,createdAt = 0} = {}) =>({
    type:'ADD_EXPENCE',
    expence:{
        id:uuid(),
        describtion,
        note,
        amount,
        createdAt
    }
});


export const removeExpence = (id=undefined) =>({
    type:'REMOVE_EXPENCE',
    id
});

export  const editExpence = (id=undefined,updates={}) =>({
    type:'EDIT_EXPENCE',
    id,
    updates
});