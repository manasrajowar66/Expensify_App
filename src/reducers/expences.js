const expencesDefaultReducer = [];

const expenceReducer = (state = expencesDefaultReducer , action ) =>{
    switch (action.type) {
        case 'ADD_EXPENCE':
            return [...state,action.expence];
        case 'REMOVE_EXPENCE':
            return state.filter(({ id })=> action.id !== id)
        case 'EDIT_EXPENCE':
            return state.map((expence)=>{
                if(expence.id === action.id)
                return {
                    ...expence,
                    ...action.updates
                };
                else
                return expence;
            })
        default:
            return state;
    }
}

export default expenceReducer;