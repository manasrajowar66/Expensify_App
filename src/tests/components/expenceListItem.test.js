import React from 'react';
import  ExpenceListItem  from '../../components/expenceListItem';
import { shallow } from 'enzyme';
import expences from '../data';

test('should render expence item',()=>{
    const wrraper = shallow(<ExpenceListItem {...expences[0]}/>);
    expect(wrraper).toMatchSnapshot();
});