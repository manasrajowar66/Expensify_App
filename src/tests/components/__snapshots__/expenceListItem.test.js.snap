// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expence item 1`] = `
<div>
  <[object Object]
    className="link-decoration"
    to="/edit/1"
  >
    <h2>
      ren
    </h2>
  </[object Object]>
  167
   - 
  0
</div>
`;
